### PR TITLE
B 20639 nts loa fetching int

### PR DESCRIPTION
**Pull request by cameroncaci**
## Summary
This ticket was rewritten after some NTS review. We're now introducing a new column to the move table to explicitly track when a move task order gets approved. Currently, the "MTO approved" timestamp is the `available_to_prime_at` database value, however in the future third party vendors may be available and NTS shipments will need a specific attribute just for MTO approval dates in order to look up the LOAs.

We are also retroactively applying this logic by setting the `approved_at` value on all existing MTOs to be equal to the `available_to_prime_at` value.

### How to test

1. Login as a new customer and create a new HHG move
2. Login as service counselor, review the order, and submit the move details
3. Login as task ordering officer and approve the requested move
4. Open your browser tools network tab -> refresh the page
5. Look for your move locator within the network tab, once found open it and verify a request URL of `ghc/v1/move/4C3XQM`
6. Read the response and verify that the attribute `approvedAt` is present and holds the same value as `availableToPrimeAt

#### Any new migrations/schema changes:

- [x] Follows our guidelines for [Zero-Downtime Deploys](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#zero-downtime-migrations).
- [x] Excel sheet was updated

Zero-downtime not necessary as this does not affect existing columns

### CODE REVIEW

**Review by traskowskycaci — COMMENTED** [review]
Working properly and db matches what I see in the response!
image
image

Not all of the approved_at values got updated when the available_to_prime_at values weren't null. Why would that be?
image

Last question is on the use of "effective." I know from your description that it's "the orders issue date or the Non-Temporary Storage (NTS) Move Task Order (MTO) approval date". But by "effective" do we mean effective for invoicing?

**Comment by cameroncaci** [comment]
> Working properly and db matches what I see in the response! image image
> 
> Not all of the approved_at values got updated when the available_to_prime_at values weren't null. Why would that be? image
> 
> Last question is on the use of "effective." I know from your description that it's "the orders issue date or the Non-Temporary Storage (NTS) Move Task Order (MTO) approval date". But by "effective" do we mean effective for invoicing?

I've begun rewording "Orders Issue Date" to be the "Effective Date" as the parameter is really only used as the filter between TAC and LOA begin and ending dates. This is mainly because NTS is about to use this endpoint and they don't use orders issue date, they use either the current day of review or the day in which the MTO was approved. Wasn't sure of another term to use

**Review by traskowskycaci — APPROVED** [review]
Updated docs look great, appreciate the extra detail on the effective date!

Also discussed the issue of not all approved_at values getting updated with Cameron. Believe it's a non-issue and could be due to timing of seed data vs migration running. Not a big deal, won't impact NTS LOA fetching functionality.

Nice work!

File: migrations/app/schema/20240729200347_add_mto_approved_at_timestamp.up.sql [hunk 1]
```diff
@@ -0,0 +1,3 @@
+ALTER TABLE moves
+ADD COLUMN approved_at timestamp with time zone;
```
Comment (danieljordan-caci) [inline migrations/app/schema/20240729200347_add_mto_approved_at_timestamp.up.sql:2]: Can you add `IF NOT EXISTS` to this migration file?

**Review by danieljordan-caci — CHANGES_REQUESTED** [review]
One little migration change and I'll stamp ya

File: migrations/app/schema/20240729200347_add_mto_approved_at_timestamp.up.sql (hunk 1 above)
Comment (cameroncaci) [reply migrations/app/schema/20240729200347_add_mto_approved_at_timestamp.up.sql]: Added

**Review by danieljordan-caci — APPROVED** [review]

**Pull request merged**